### DISCUSSION TITLE
How do you install the cert to vaultwarden?

### DISCUSSION THREAD

**Discussion by danieldotkim** [dani-garcia/vaultwarden#4213] [Q&A]
I am using an AWS EC2 instance to host VaultWarden, and I use CloudFlare for DNS.
I followed this instruction from CertBot: https://certbot.eff.org/instructions?ws=other&os=ubuntufocal&tab=wildcard.
It generated the cert and private key in my /etc/letsencrypt/live/domain.com/, but I am unsure of how I can "install" the certificate as it instructs me in CertBot.

The "[Getting SSL/TLS Certificates](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS#getting-ssltls-certificates)" section does tell me to use LetsEncrypt or CloudFlare, and I am able to generate the certs without any issue. However, I am unsure of how to implement the certs to my domain and get it working on CloudFlare. The VaultWarden is running, and it is reachable directly through the IP address of my AWS EC2 instance.

I can only navigate to VaultWarden through HTTP despite having the certificate and I am stumped on how to implement the certificate .

If I enable the proxy option for CloudFlare DNS A records, it allows me to access VaultWarden, but doesn't that defeat the purpose of setting up the certificate manually?

Any help would be greatly appreciated.

**Comment by tessus** [comment]
This all depends on several things. You mentioned that you are using Cloudlare for DNS. Is that really all? Many people use CF for TLS termination.

You also mentioned that you access vw via the IP address of the EC2 instance. This still does not tell us whether you use vw in a container or binary (manually or via a package), nor if you use a reverse proxy or not.

e.g. I use Apache as a reverse proxy and the TLS cert is setup in that virtual host.

**Comment by danieldotkim** [comment]
Sorry for the lack of information. I am not using a reverse proxy at the moment.

I used docker run to run the VaultWarden:
docker run -d --name vaultwarden \
  -e DOMAIN=http://domain.com \
  -e ADMIN_TOKEN=placeholder-token \
  -v /vw-data/:/data \
  -p 80:80 \
  vaultwarden/server:latest

Unrelated, but config.json does not appear when running the container with the specified ADMIN_TOKEN environmental variable when running the container.

**Reply by tessus** [reply]
I think at one point it was said that TLS via rocket might not be a good idea. I can't recall why. Maybe because it was experimental at that time.

If this is no longer the case case, you can just set the cert in the vaultwarden config file and use port 443.

Otherwise just setup a reverse proxy with the TLS cert and point to `localhost:80`. Btw, running something on port 80 on the host might not be the best idea anyway, unless you have nothing else running on that box.
I think you have to decide on your architecture first.

**Comment by danieldotkim** [comment]
Just tried running using docker-compose method with Caddy. It seems to work this way and my issue is resolved.

**Reply by tessus** [reply]
ok. it seems you found your architecture....
